(In Progress)[Plugin] Backup: Update Backup Segments section content.

nunyvega opened the pull request:
Modify the content shown on the Backup Segments section, right under the main content. 

#### TO DO:
- [x] Modify the cloud backups section and add link to backups
- [x] Remove the Upgrade to `Realtime prompt`
- [ ] Add the My Plan section

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
Add the next modifications to the Backup Segments section:
- Change text on this block.
- Add link to see all backups on Jetpack Cloud.
- Remove the message to upgrade to realtime.

#### Jetpack product discussion
**Project Asana card:**
1201228005200946-as-1201373372830517

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
On a site running the code on this PR:
1- Create a site running the code of the plugin of this PR
2- Open the Jetpack Backup admin section
3- Check that the text under the main section matches the one on this screenshot:
![image](https://user-images.githubusercontent.com/16329583/141999257-e493c2ef-991c-4d76-82b4-677ecfc37029.png)
4- Confirm that the `See all your backups` link takes to the Backups section of your site on Jetpack Cloud.

jwebbdev commented:
@nunyvega Can we pull the logic for fetching the purchases into MyPlan?  I think eventually we'll extract this if other plugins will be using it.  We don't yet have a good solution for the general end points, like `/site/purchase` but this will probably end up in its own package eventually.

nunyvega commented:
To complete the rebase I had to run `pnpm install`  in `jetpack/projects/plugins/boost/tests/e2e`  and to run `git commit --no-verify`. 

This second one was because ` jetpack/vendor/sirbrillig/phpcs-changed/PhpcsChanged/PhpcsMessages.php` was throwing the exception:
```
Uncaught Exception: Failed to decode phpcs JSON: 
Fatal error: Uncaught TypeError: vsprintf(): Argument #2 ($values) must be of type array, string given in /Users/nunyvega/Documents/a8c-projects/jetpack/vendor/squizlabs/php_codesniffer/src/Files/File.php:1056
```

nunyvega commented:
@jwebbdev thanks for the recommendation, it makes sense.  I moved the logic to fetch the purchases into the component and changed the component a bit so it can be easier to make it work for other types of purchases.

jwebbdev commented on projects/plugins/backup/src/js/components/MyPlan.js:
```diff
@@ -0,0 +1,77 @@
+function getPurchasesList( purchases, purchaseType ) {
+	const purchasesList = [];
+	purchases.forEach( purchase => {
+		if ( purchaseType === 'backup' ) {
```
Since we just handle `backup` as a `purchaseType` maybe there should be a default of `all` and we just return the purchases without filtering.  At the moment you have to pass `backup` for a non-empty result.

nunyvega replied:
Good idea, I just added it :)

jwebbdev commented on projects/plugins/backup/src/js/components/MyPlan.js:
```diff
@@ -0,0 +1,77 @@
+	isJetpackBackup,
+	isJetpackBundle,
+	isJetpackLegacyPlan,
+} from '../../../../jetpack/_inc/client/lib/plans/constants';
```
@Automattic/jetpack-crew - Can anyone explain why this seems to work?  I've tested it on a JN site without the Jetpack plugin and would expect it to have a problem.  Is there something in the tooling that pulls it in?

oskosk replied:
Webpack just goes, follows that path, opens the file, copies the content into its own bah, transpiles it and sticks it into the final bundle where root paths are no longer a thing basically. Regular webpack behaviour 👍

jwebbdev commented on projects/plugins/backup/src/js/components/MyPlan.js:
```diff
@@ -0,0 +1,77 @@
+			},
+			() => {
+				setSitePurchasesLoaded( true );
+				setSitePurchasesError( 'Failed to fetch site purchases' );
```
Need to wrap in translation function

nunyvega replied:
Done ✅

jwebbdev commented on projects/plugins/backup/src/js/components/MyPlan.js:
```diff
@@ -0,0 +1,90 @@
+
+	filteredPurchases.forEach( purchase => {
+		purchasesList.push(
+			<>
```
We can use [Fragment](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-element/#fragment) here.

jwebbdev commented on projects/plugins/backup/src/php/class-jetpack-backup.php:
```diff
@@ -158,8 +158,17 @@ public function register_rest_routes() {
+		// Get information on site products.
+		register_rest_route(
+			'jetpack/v4',
+			'/site/purchases',
```
We could use a unique route here, with the intention to revert to this, in a later version, once the functionality is extracted into a common package.  Currently we're using the same route as the jetpack plugin, and depending on which route gets called while both plugins are active a change to either could break the other.

jwebbdev commented on projects/plugins/backup/src/js/components/Admin.js:
```diff
@@ -137,26 +138,29 @@ const Admin = () => {
+							</p>
+							<MyPlan
+								purchaseType={ 'backup' }
+								redirectUrl={ getRedirectUrl( 'backup-plugin-my-plan', { site: domain } ) }
```
This sent me to `https://wordpress.com/plans/my-plan/<site>?site=<site>`. WordPress doesn't handle backup-only sites well, so we should direct to `https://cloud.jetpack.com/purchases/subscriptions/<site>` instead I think.

nunyvega replied:
Done! I changed it in the redirect manager

jwebbdev commented:
Tested with the new redirect and looks good.
